src/App.tsx

```diff
@@ -100,12 +71,56 @@ function App() {
       return;
     }
 
-    toast.promise(sendZip(archive), {
-      loading: "Uploading",
+    toast.promise(sendContent(archive), {
+      loading: "Uploading..",
       success: handleUploadSuccess,
       error: handleUploadError,
     });
-  };
+  }
+
+  const handleUpload = async (files: File[]) => {
+    let converter: MediaConverter | undefined;
+
+    for (const file of files) {
+      if (file.size == 0) {
+        toast.error(handleUploadError("Invalid file."));
+        break;
+      }
+
+      if (isZipFile(file)) {
+        handleZipUpload(file);
+        continue;
+      }
+
+      if (isImageFile(file))
+        converter = imageConverter;
+      else if (isFontFile(file))
+        converter = fontConverter;
+
+      if (converter === undefined) {
+        toast.error(handleUploadError("Invalid file type."));
+        break;
+      }
+
+      toast.promise(
+        converter.convert(files.map((file: File) => ({filepath: file.name, data: file}))),
+        {
+          loading: "Uploading..",
+          success: (files: MediaFile[]) => {
+            playSuccess();
+            const zip = new JSZip();
+
+            for (const file of files) {
+              zip.file(file.filepath, file.data);
+            }
+
+            zip.generateAsync({type: "blob"}).then((blob: Blob) => downloadBlob(blob));
+            return "Downloaded.";
+          },
+          error: handleUploadError
+        });
```

Comment:
Suggestion: for readibility sake, I'd rather not use `continue` and `break` statements, but split this block of code into methods/functions and use an `if .. else` statement
```typescript
for (const file of files) {
  if (file.size === 0) {
    break; //here it's okay, because it's just a clear failsafe
  } else if (isZipFile(file)) {
    handleZipUpload(file);
  } else {
    handleMediaUpload(file);
  }
}
```